src/Lightning.ts

```diff
@@ -290,6 +291,23 @@ export const LightningMixin = (superclass) =>
         params,
       })
 
+      if (
+        yamlConfig.twoFactor?.enabled &&
+        this.user.twoFactor.secret &&
+        tokens > this.user.twoFactor.threshold
+      ) {
+        if (!token) {
+          throw new TwoFactorError("Need a 2FA code to proceed with the payment", {
+            logger: lightningLogger,
+          })
+        }
+
+        UserWallet.validate2fa({
```

Comment:
I guess the assumption here is that we throw if the token is not correct. but I don't think it's clear by reading the code. 

specially at the function return a boolean.

I think: either we don't return a boolean and have a `throwIfNotValid()` or we use the boolean and throw the error locally.

Reply:
I'm going with the former option